src/charm.py

```diff
@@ -1,427 +1,269 @@
-                endpoint_url += f".{obj_storage['namespace']}"
-            if "port" in obj_storage and len(obj_storage["port"]) > 0:
-                endpoint_url += f":{obj_storage['port']}"
-
-        return endpoint_url
-
-
-class CheckFailedError(Exception):
-    """Raise this exception if one of the checks in main fails."""
-
-    def __init__(self, msg, status_type=StatusBase):
-        super().__init__()
-
-        self.msg = str(msg)
-        self.status_type = status_type
-        self.status = status_type(self.msg)
-
-
-def _gen_obj_storage_endpoint_url(obj_storage):
-    """Generate object storage endpoint URL."""
-    return f"http://{obj_storage['service']}.{obj_storage['namespace']}:{obj_storage['port']}"
-
-
-def _b64_encode_dict(d):
-    """Returns the dict with values being base64 encoded."""
-    # Why do we encode and decode in utf-8 first?
-    return {k: b64encode(v.encode("utf-8")).decode("utf-8") for k, v in d.items()}
-
-
-def _minio_credentials_dict(obj_storage):
-    """Returns a dict of minio credentials with the values base64 encoded."""
-    minio_credentials = {
-        "AWS_ENDPOINT_URL": _gen_obj_storage_endpoint_url(obj_storage),
-        "AWS_ACCESS_KEY_ID": obj_storage["access-key"],
-        "AWS_SECRET_ACCESS_KEY": obj_storage["secret-key"],
-        "USE_SSL": str(obj_storage["secure"]).lower(),
-    }
-    return _b64_encode_dict(minio_credentials)
-
-
-def _seldon_credentials_dict(obj_storage):
-    """Returns a dict of seldon init-container object storage credentials, base64 encoded."""
-    credentials = {
-        "RCLONE_CONFIG_S3_TYPE": "s3",
-        "RCLONE_CONFIG_S3_PROVIDER": "minio",
-        "RCLONE_CONFIG_S3_ACCESS_KEY_ID": obj_storage["access-key"],
-        "RCLONE_CONFIG_S3_SECRET_ACCESS_KEY": obj_storage["secret-key"],
-        "RCLONE_CONFIG_S3_ENDPOINT": _gen_obj_storage_endpoint_url(obj_storage),
-        "RCLONE_CONFIG_S3_ENV_AUTH": "false",
-    }
-    return _b64_encode_dict(credentials)
+        if not ((obj_storage := interfaces["object-storage"]) and obj_storage.get_data()):
+            raise ErrorWithStatus("Waiting for object-storage relation data", WaitingStatus)
 
+        try:
+            obj_storage = list(obj_storage.get_data().values())[0]
+        except Exception as e:
+            raise ErrorWithStatus(
+                f"Unexpected error unpacking object storage data - data format not "
+                f"as expected. Caught exception: '{str(e)}'",
+                BlockedStatus,
+            )
 
-def _db_secret_dict(mysql):
-    """Returns a dict of db-secret credential data, base64 encoded."""
-    db_secret = {
-        "DB_ROOT_PASSWORD": mysql["root_password"],
-        "MLFLOW_TRACKING_URI": f"mysql+pymysql://root:{mysql['root_password']}@{mysql['host']}"
-        f":{mysql['port']}/{mysql['database']}",
-    }
-    return _b64_encode_dict(db_secret)
+        return obj_storage
+
+    def _get_env_vars(self, relational_db_data, object_storage_data):
```

Comment:
This method is static - it doesn't need `self` so can either be a `@staticmethod` or a helper